From fc404dba8d7e4e3583d81e08e542d91d459496f1 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Tue, 25 Feb 2025 17:38:33 +0100
Subject: [PATCH 1607/1695] dts: stm32: remove interrupt for RTC on stm32mp13

Remove the RTC interrupt in STM32MP13 as RTC is not secured by default
for STM32MP13 boards.

This interruption should be managed by OP-TEE only if RTC is secured,
and when access by non secure world is provided by CFG_RTC_PTA.

This patch avoid issue when EXTI are secured in OP-TEE when they are
registered by EXTI driver.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9627606c8fe2dbd878b8f091f3815f6bbccbe612 dts: stm32: remove interrupt for RTC on stm32mp13]
Fixes: 090d1937c3a7 ("dts: stm32: fix RTC node at STM32MP1 SoC level")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I76ec4eb1d7a1903c4da9b425a8dcf47f55830078
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/439971
Tested-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 1 -
 1 file changed, 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 4e1dbd979..8cec39dc9 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -399,7 +399,6 @@
 			reg = <0x5c004000 0x400>;
 			clocks = <&rcc RTCAPB>, <&rcc RTC>;
 			clock-names = "pclk", "rtc_ck";
-			interrupts-extended = <&exti 19 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		bsec: efuse@5c005000 {
-- 
2.47.2

